rmw_fastrtps_shared_cpp/src/rmw_subscription.cpp

```diff
@@ -53,42 +49,30 @@ __rmw_destroy_subscription(
     return RMW_RET_ERROR;
   }
 
-  if (!subscription) {
-    RMW_SET_ERROR_MSG("subscription handle is null");
-    return RMW_RET_ERROR;
-  }
-
-  if (subscription->implementation_identifier != identifier) {
-    RMW_SET_ERROR_MSG("node handle not from this implementation");
-    return RMW_RET_ERROR;
-  }
```

Comment:
@ivanpauno same about not removing the two checks above.

Reply:
Moved to https://github.com/ros2/rmw_fastrtps/blob/96ad1d3ad6c764dd45dadf3fc3a7592fb51c58c4/rmw_fastrtps_shared_cpp/src/subscription.cpp#L47-L54.

Reply:
Same as before, `subscription` is being used in https://github.com/ros2/rmw_fastrtps/pull/312/files#diff-57b111a3ec82a425bffeb7e3633c762eR53 already.

Reply:
See https://github.com/ros2/rmw_fastrtps/pull/312/commits/9fd7fe605ca749201371a4636274c2bfdb8ec1d0